client/platform/desktop/backend/serializers/viame.ts

```diff
@@ -0,0 +1,273 @@
+  key = '',
+) {
+  const geoFeature: GeoJSON.Feature<TrackSupportedFeature, GeoJSON.GeoJsonProperties> = {
+    type: 'Feature',
+    properties: { key },
+    geometry: {
+      type,
+      coordinates: [],
+    },
+  };
+  if (type === 'Polygon') {
+    geoFeature.geometry.coordinates = [coords];
+  } else if (type === 'Point') {
+    [geoFeature.geometry.coordinates] = coords;
+  } else {
+    geoFeature.geometry.coordinates = coords;
+  }
+  return geoFeature;
+}
+
+function _parseRow(row: string[]) {
+  // Create empty feature collection
+  const geoFeatureCollection:
+    GeoJSON.FeatureCollection<TrackSupportedFeature, GeoJSON.GeoJsonProperties> = {
+      type: 'FeatureCollection',
+      features: [],
+    };
+  let attributes: StringKeyObject | undefined;
+  const trackAttributes: StringKeyObject = {};
+  const cpStarti = 9; // Confidence pairs start at i=9
+  const confidencePairs: ConfidencePair[] = row
+    .slice(cpStarti, row.length)
+    .map((_, j) => {
+      if (j % 2 !== 0) {
+        // Filter out ODDs
+        return ['', 0] as ConfidencePair;
+      }
+      const i = j + cpStarti;
+      if ((i + 1) < row.length && row[i] && row[i + 1] && !row[i].startsWith('(')) {
+        return [row[i], parseFloat(row[i + 1])] as ConfidencePair;
+      }
+      return ['', 0] as ConfidencePair;
+    })
+    .filter((val) => val[0] !== '');
+  const headTail: [number, number][] = [];
+  const start = 9 + (confidencePairs.length * 2);
+  row.slice(start).forEach((value) => {
+    /* Head */
+    const head = getCaptureGroups(HeadRegex, value);
+    if (head !== null) {
+      headTail[0] = [parseFloat(head[1]), parseFloat(head[2])];
+      geoFeatureCollection.features.push(_createGeoJsonFeature('Point', [headTail[0]], 'head'));
+    }
+
+    /* Tail */
+    const tail = getCaptureGroups(TailRegex, value);
+    if (tail !== null) {
+      headTail[1] = [parseFloat(tail[1]), parseFloat(tail[2])];
+      geoFeatureCollection.features.push(_createGeoJsonFeature('Point', [headTail[1]], 'tail'));
+    }
+
+    /* Detection Attribute */
+    const attr = getCaptureGroups(AttrRegex, value);
+    if (attr !== null) {
+      if (attributes === undefined) attributes = {};
+      attributes[attr[1]] = _deduceType(attr[2]);
+    }
+
+    /* Track Attribute */
+    const trackattr = getCaptureGroups(TrackAttrRegex, value);
+    if (trackattr !== null) {
+      trackAttributes[trackattr[1]] = _deduceType(trackattr[2]);
+    }
+
+    /* Polygon */
+    const poly = getCaptureGroups(PolyRegex, value);
+    if (poly !== null) {
+      const coords: number[][] = [];
+      const polyList = poly[2].split(' ');
+      polyList.forEach((coord, j) => {
+        if (j % 2 === 0) {
+          // Filter out ODDs
+          if (polyList[j + 1]) {
+            coords.push([parseFloat(coord), parseFloat(polyList[j + 1])]);
+          }
+        }
+      });
+      geoFeatureCollection.features.push(_createGeoJsonFeature('Polygon', coords));
+    }
+  });
```

Comment:
Just a reminder to eventually add in some better checking here for `poly` being an array with an even number of coordinates.  (I think that should be a requirement)

Reply:
Captured in https://github.com/VIAME/VIAME-Web/issues/444